packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -383,6 +400,14 @@ class _TestRecordingPaintingContext implements PaintingContext {
   }
 
   @override
+  void pushClipRect(bool needsCompositing, Offset offset, Rect clipRect, PaintingContextCallback painter) {
+    canvas.save();
```

Comment:
we should document somewhere (in this file's docs) that we're implying a save/restore pair when you call pushClipRect.